src/lib/storage/value_segment/value_segment_iterable.hpp

```diff
@@ -98,8 +110,16 @@ class ValueSegmentIterable : public PointAccessibleSegmentIterable<ValueSegmentI
       ++_chunk_offset;
     }
 
+    void advance(std::ptrdiff_t n) {
```

Comment:
`distance_to` I get, but what is "advance" necessary for?

Reply:
https://www.boost.org/doc/libs/1_65_0/libs/iterator/doc/iterator_facade.html#iterator-facade-operations


I ended up not using it, but I believe both @FabianWiebe and @Bensk1 used it at some point.

Reply:
If this is definitely going to be used by an impending PR, keep it in, otherwise it's obsolete code and should be removed.